### PR TITLE
Restructure the js app

### PR DESCRIPTION
**Pull request by woss**
remove the common code and merge it to the frontend

### CODE REVIEW

File: frontend/src/persist/Persistent.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ export class Persistent<Data> {
     this.key = key;
     this.onChange = onChange;
 
-    const stored = window.localStorage.getItem(key) as Maybe<Stringified<Data>>;
+    const stored = window.localStorage.getItem(key) as any;
```
Comment (chevdor) [inline frontend/src/persist/Persistent.ts:16]: Why ?

File: frontend/src/state.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
-import { Types, Maybe, SortedCollection } from '@dotstats/common';
+import { Types, Maybe, SortedCollection } from './common';
 import { Column } from './components/List';
 
-export const PINNED_CHAIN = 'Kusama';
+export const PINNED_CHAIN = 'Kusama CC3';
```
Comment (chevdor) [inline frontend/src/state.ts:4]: AFAIK, the network was renamed recently from `Kusama CC3` to `Kusama` so this change does not look ok.

File: frontend/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  "exclude": [
+    "node_modules",
+    "build"
+  ],
+  "include": [
+    "src/**/*.ts",
+    "src/**/*.tsx",
+    "./declarations/**/*.d.ts"
+  ]
+}
```
Comment (chevdor) [inline frontend/tsconfig.json:38]: Missing EOL

File: frontend/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    "jsx": "react",
+    "rootDir": "src",
+    "moduleResolution": "node",
+    "allowSyntheticDefaultImports": true,
+    "noEmitOnError": false,
+    "pretty": true,
+    "forceConsistentCasingInFileNames": false,
+    "noErrorTruncation": true,
+    "noImplicitReturns": true,
+    "noImplicitAny": false,
```
Comment (chevdor) [inline frontend/tsconfig.json:22]: This used to be `true`. Going back to `false` without an excellent reason would be a regression.

File: frontend/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,38 @@
+{
+  "compilerOptions": {
+    "baseUrl": ".",
+    "outDir": "build",
+    "module": "esnext",
```
Comment (chevdor) [inline frontend/tsconfig.json:5]: Used to be `commonjs`. Sure of that change?

File: frontend/tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    "target": "ES2017",
+    "lib": [
+      "es6",
+      "dom"
+    ],
+    "sourceMap": true,
+    "allowJs": true,
```
Comment (chevdor) [inline frontend/tsconfig.json:12]: Do we really need `allowJs` to be `true` now ?

File: frontend/tsconfig.json [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    "moduleResolution": "node",
+    "allowSyntheticDefaultImports": true,
+    "noEmitOnError": false,
+    "pretty": true,
+    "forceConsistentCasingInFileNames": false,
+    "noErrorTruncation": true,
+    "noImplicitReturns": true,
+    "noImplicitAny": false,
+    "noImplicitThis": true,
+    "strictNullChecks": false,
```
Comment (chevdor) [inline frontend/tsconfig.json:24]: Used to be true. Why this change?

File: frontend/tsconfig.json [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+    "suppressImplicitAnyIndexErrors": true,
+    "resolveJsonModule": true,
+    "noUnusedParameters": false
```
Comment (chevdor) [inline frontend/tsconfig.json:27]: From what I see in the code, the frontend was coded the Rust way so unused are marked with `_`.  So instead of allowing unused, we should only allow unused that start with a `_`. I think the old code should pass this way and we prevent regressions in new code.

File: frontend/src/components/List/Column.tsx [hunk 9]
```diff
@@ -58,7 +58,6 @@ import turingIcon from '../../icons/turingnetwork.svg';
 import dothereumIcon from '../../icons/dothereum.svg';
 import katalchainIcon from '../../icons/katalchain.svg';
 import bifrostIcon from '../../icons/bifrost.svg';
-import totemIcon from '../../icons/totem.svg';
```
[the remark this answers is not in the document]
Comment (chevdor) [reply frontend/src/components/List/Column.tsx:61]: This was added a few days ago in:
- #236
- #239

File: README.md [hunk 10]
```diff
@@ -20,35 +22,45 @@ cargo build --release
...
 BIND=0.0.0.0 telemetry
 ```
 
 ### Terminal 2 - Frontend
-```
-yarn start:frontend
+
+```sh
+cd frontend
+yarn
```
[the remark this answers is not in the document]
Comment (woss) [reply README.md:40]: as the whole thing moved away from the yarn `workspaces`  we need to manually install the deps in the frontend itself.  i don't see why a user cannot just co to the `frontend` and `yarn && yarn start`. But for the sake of the fast start there is a script in the root `start` which starts the frontend.

File: frontend/package.json [hunk 11]
```diff
@@ -25,8 +25,8 @@
     "blakejs": "^1.1.0",
     "husky": "^4.2.3",
     "lint-staged": "^10.1.0",
-    "react": "16.4.0",
-    "react-dom": "16.4.2",
+    "react": "^16.13.1",
```
[the remark this answers is not in the document]
Comment (woss) [reply frontend/package.json:28]: no we cannot, then we need to extract ConnectionInput which is built using the hooks, and hooks are not available in the `16.4.2`

File: frontend/src/components/ConnectionInput/ConnectionInput.tsx [hunk 12]
```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
+
```
[the remark this answers is not in the document]
Comment (woss) [reply frontend/src/components/ConnectionInput/ConnectionInput.tsx:2]: checks passing, so yes linter is ok with that. grouping ?  maybe next time or some automatic way of doing it

File: frontend/tsconfig.json (hunk 4 above)
Comment (woss) [reply frontend/tsconfig.json:22]: because of this `substrate-telemetry/frontend/src/AfgHandling.ts
(16,10): 'receivedAuthoritySet', which lacks return-type annotation, implicitly has an 'any' return type.`

Comment (woss) [reply frontend/tsconfig.json:22]: and potentially other issues when you solve that. setting that to false is nothing major. if we put that back code will break. also deleting the yarn.lock from frontend will break it even on master. try it.

Comment (woss) [reply frontend/tsconfig.json:22]: even before all my big prs

File: frontend/tsconfig.json (hunk 6 above)
Comment (woss) [reply frontend/tsconfig.json:12]: back to false

File: frontend/tsconfig.json (hunk 8 above)
Comment (woss) [reply frontend/tsconfig.json:27]: the moment i opened the project i saw a strongly typed background of a person who did this. the flag was deprecated.

File: frontend/tsconfig.json (hunk 4 above)
Comment (chevdor) [reply frontend/tsconfig.json:22]: "Implicit any" issues are only showing up in ConnectionInput.tsx => specifying the 2 missing types
```
  function handleChange(e: React.ChangeEvent<HTMLInputElement>) {... }
  function handleSubmit(e: SyntheticEvent) { ... }
```

File: frontend/tsconfig.json (hunk 7 above)
Comment (chevdor) [reply frontend/tsconfig.json:24]: It requires only a simple fix in `ConnectionInput.tsx` to be set to true

File: frontend/tsconfig.json (hunk 8 above)
Comment (chevdor) [reply frontend/tsconfig.json:27]: Setting `"noUnusedParameters": true` requires the following changes:
- AfgHandling.ts:L116
- AfgHandling.ts:L139
- AfgHandling.ts:L195
- App.tsx:L194
- Consensus.tsx:360
- Sparkline.tsx:82

File: frontend/package.json (hunk 11 above)
[the remark this answers is not in the document]
Comment (chevdor) [reply frontend/package.json:28]: 👍

File: README.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (chevdor) [reply README.md:40]: 👍

File: frontend/src/components/ConnectionInput/ConnectionInput.tsx (hunk 12 above)
[the remark this answers is not in the document]
Comment (chevdor) [reply frontend/src/components/ConnectionInput/ConnectionInput.tsx:2]: Yeah.. that hurts my eyes a bit but I admit this is a detail.
That can be fixed later.

File: frontend/src/components/List/Column.tsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (chevdor) [reply frontend/src/components/List/Column.tsx:61]: This issue should not be forgotten

File: frontend/tsconfig.json (hunk 6 above)
Comment (chevdor) [reply frontend/tsconfig.json]: 👍

File: frontend/tsconfig.json (hunk 5 above)
Comment (woss) [reply frontend/tsconfig.json]: back to commonjs

File: frontend/src/Connection.ts [hunk 13]
```diff
@@ -17,12 +10,11 @@ import {
...
 import { AfgHandling } from './AfgHandling';
-import { VIS_AUTHORITIES_LIMIT } from '../../frontend/src/components/Consensus';
+import { VIS_AUTHORITIES_LIMIT } from './components/Consensus';
 import { Column } from './components/List';
+import { ACTIONS } from './common/feed';
 
-const { Actions } = FeedMessage;
-
-const TIMEOUT_BASE = (1000 * 5) as Types.Milliseconds; // 5 seconds
+const TIMEOUT_BASE = (1000 * 5) as Types.Milliseconds; // 3 seconds
```
Comment (chevdor) [inline frontend/src/Connection.ts:17]: 5s

File: frontend/src/components/List/Column.tsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (maciejhirsz) [reply frontend/src/components/List/Column.tsx:61]: Yeah, this will need to be re-added.

File: frontend/src/components/Map/Location.tsx [hunk 14]
```diff
@@ -91,7 +91,7 @@ export class Location extends React.Component<Location.Props, Location.State> {
       city,
     } = this.props.node;
 
-    let validatorRow = null;
+    let validatorRow = <div />;
```
Comment (maciejhirsz) [inline frontend/src/components/Map/Location.tsx:94]: Any reason for this change? This will insert an empty `div` inside a `tbody`, which doesn't quite make sense.

File: frontend/src/state.ts (hunk 2 above)
Comment (maciejhirsz) [reply frontend/src/state.ts:4]: Yup :point_up:

File: frontend/src/persist/Persistent.ts (hunk 1 above)
Comment (maciejhirsz) [reply frontend/src/persist/Persistent.ts:16]: I'd definitely keep `Maybe<Stringified<Data>>` here.

File: frontend/src/components/Map/Location.tsx (hunk 14 above)
Comment (chevdor) [reply frontend/src/components/Map/Location.tsx:94]: This comes from activating more null check rules.
